src/state/schedule-data/schedule-errors/schedule-errors.reducer.ts

```diff
@@ -2,28 +2,21 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 import _ from "lodash";
+import { createAction, createReducer } from "@reduxjs/toolkit";
 import { GroupedScheduleErrors, ScheduleError } from "./schedule-error.model";
-import { ActionModel } from "../../../utils/action.model";
-import { ScheduleActionType } from "../schedule.actions";
 
 export enum ScheduleErrorActionType {
   UPDATE = "updateScheduleError",
 }
-
-export function scheduleErrorsReducer(
-  state: GroupedScheduleErrors = {},
-  action: ActionModel<ScheduleError[]>
-): GroupedScheduleErrors {
-  switch (action.type) {
-    case ScheduleErrorActionType.UPDATE:
+export const updateScheduleErrors = createAction<ScheduleError[]>("schedule/updateErrors");
+export const cleanScheduleErrors = createAction("schedule/cleanErrors");
+export const scheduleErrorsReducer = createReducer({} as GroupedScheduleErrors, (builder) => {
+  builder
+    .addCase(updateScheduleErrors, (state, action) => {
       if (!action.payload) action.payload = [];
       const errors = _.groupBy(action.payload, (item) => item.kind);
       return errors;
-
-    case ScheduleActionType.CLEAN_ERRORS:
-      // In case if new schedule is added we should remove error-list, that previously existed
-      return {};
-    default:
-      return state;
-  }
-}
+    })
+    .addCase(cleanScheduleErrors, (state, action) => null)
```

Comment:
Dlaczego `null`, a nie `{}`?

Wcześniej był zwracany pusty obiekt ponieważ `scheduleErrors` - to słownik i tym miejscu my go zamienialiśmy na pusty słownik.

Teraz jak zwracamy `null`, to w stanie też będzie `null`.

I wtedy w momencie kiedy my spróbujemy dostać się do jakiegoś klucza na tym null do dostaniemy błąd.

Reply:
Ok.